src/zino/snmp/netsnmpy_backend.py

```diff
@@ -24,18 +25,43 @@
     MibObject,
     NoSuchInstanceError,
     NoSuchObjectError,
+    SNMPBackendError,
+    SNMPBackendVersionError,
     SNMPVarBind,
     SparseWalkResponse,
 )
 
 _log = logging.getLogger(__name__)
 
 
-def get_new_snmp_engine():
-    """This is only here because the pysnmp backend provides it.  It needs to be
-    gone, but we need to figure out how it is used outside this module, if at all.
-    """
-    raise NotImplementedError
+def init_backend():
```

Comment:
Why is `get_new_snmp_engine` missing from this file? If it is not necessary, maybe add a comment inside the function in src/zino/snmp/pysnmp_backend.py

Reply:
Because it became redundant, as was expected when the second back-end was first added.

At this point, it is only re-used outside of this module by the PySNMP specific `trapd` backend, but is not part of the public interface expected from an SNMP back-end by the rest of Zino.

I've expounded slightly on this in the docstring, see new commit.